src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs

```diff
@@ -1338,5 +1377,54 @@ private static object SearchIntrinsicTable(Hashtable table, Type callingType)
 
             return hashEntry;
         }
+
+        /// <summary>
+        /// Searches the intrinsic converter dictionary for a match with the object type.
+        /// The strongly-typed dictionary maps object types to converter data objects which lazily
+        /// creates (and caches for re-use, where applicable) converter instances.
+        /// </summary>
+        private static TypeConverter GetIntrinsicTypeConverter(Type callingType)
+        {
+            TypeConverter converter;
+
+            // We take a lock on this dictionary. Nothing in this code calls out to
+            // other methods that lock, so it should be fairly safe to grab this lock.
+            lock (IntrinsicTypeConverters)
+            {
+                if (!IntrinsicTypeConverters.TryGetValue(callingType, out IntrinsicTypeConverterData converterData))
+                {
+                    if (callingType.IsEnum)
+                    {
+                        converterData = IntrinsicTypeConverters[typeof(Enum)];
+                    }
+                    else if (callingType.IsArray)
+                    {
+                        converterData = IntrinsicTypeConverters[typeof(Array)];
+                    }
+                    else if (Nullable.GetUnderlyingType(callingType) != null)
```

Comment:
You really only need to know if the type is Nullable, which can be achieved with:

`(callingType.IsGenericType && callingType.GetGenericTypeDefinition() == typeof(Nullable<>)`. Getting the underlying type is unnecessary work

Reply:
`Nullable.GetUnderlyingType` also checks for `IsGenericTypeDefinition`: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Nullable.cs#L120-L124

Reply:
> Nullable.GetUnderlyingType also checks for IsGenericTypeDefinition

Good catch. Another good reason not to use it. Previously if someone would pass `typeof(Nullable<>)` into this method, the previous implementation would find the right entry. But the change to use `Nullable.GetUnderlyingType` would break this behavior.